index.js

```diff
@@ -324,16 +338,32 @@ morgan.token('response-time', function getResponseTimeToken(req, res, digits) {
  * current date
  */
 
+function formatTimezone(date, timezone) {
+  // Takes a date object and formats it as:
+  // Thu Oct 19 2017 12:35:19 GMT+0530 (IST)
+  const tzDate = moment(date).tz(timezone)
+}
+
 morgan.token('date', function getDateToken(req, res, format) {
+  format = format.split(',');
+  var dateFormat = format[0];
+  var timezone = format[1];
+
   var date = new Date();
 
-  switch (format) {
-    case 'clf':
-      return clfdate(date);
+  switch (dateFormat) {
+    case: 'utc':
+      return date.toUTCString();
     case 'iso':
       return date.toISOString();
     default:
-      return date.toUTCString();
+      if (timezone) date = formatTimezone(date, timezone);
+      switch (dateFormat) {
+        case 'clf':
+          return clfdate(date);
+        default:
+          return date.toString();
+      }
```

Comment:
switching the `default` case handling from UTC to the JS default, this seems more appropriate